src/cryptography/hazmat/primitives/asymmetric/dh.py

```diff
@@ -87,9 +87,6 @@ def __init__(self, p, g, q=None):
         if q is not None and not isinstance(q, six.integer_types):
             raise TypeError("q must be integer or None")
 
-        if q is None and g not in (2, 5):
-            raise ValueError("DH generator must be 2 or 5")
```

Comment:
Do you convert the DHParameterNumbers to a DHParameters via `parameters`? I guess we don't validate anything on that path so you can make a parameters and then call `generate_private_key`.